src/common/core/mixincore.js

```diff
@@ -0,0 +1,793 @@
+            for (i = 0; i < names.length; i++) {
+                meta.attributes[names[i]] = self.getAttributeMeta(node, names[i]);
+            }
+
+            //pointers
+            names = self.getValidPointerNames(node);
+            for (i = 0; i < names.length; i++) {
+                meta.pointers[names[i]] = self.getPointerMeta(node, names[i]);
+                convertRuleToItemizedArraySet(meta.pointers[names[i]]);
+            }
+
+            //sets
+            names = self.getValidSetNames(node);
+            for (i = 0; i < names.length; i++) {
+                meta.pointers[names[i]] = self.getPointerMeta(node, names[i]);
+                convertRuleToItemizedArraySet(meta.pointers[names[i]]);
+            }
+
+            //aspects
+            names = self.getValidAspectNames(node);
+
+            for (i = 0; i < names.length; i++) {
+                meta.aspects[names[i]] = self.getAspectMeta(node, names[i]);
+            }
+
+            //constraints
+            names = self.getConstraintNames(node);
+            for (i = 0; i < names.length; i++) {
+                meta.constraints[names[i]] = self.getConstraint(node, names[i]);
+            }
+
+            return meta;
+        };
+
+        this.getOwnJsonMeta = function (node) {
+            // We have to extend the meta definition by the mixins, but only if we talk about the own rule-set.
+            var jsonMeta = innerCore.getOwnJsonMeta(node),
+                mixins = this.getMixinPaths(node);
+
+            if (mixins.length > 0) {
+                jsonMeta.mixins = mixins;
+            }
+
+            return jsonMeta;
+        };
+
+        this.getValidChildrenPaths = function (node) {
+            return getValidNames(node, innerCore.getValidChildrenPaths, []);
+        };
+
+        this.getChildrenMeta = function (node) {
+            var ruleHolders = getAllMatchingRuleHolders(node, 'containment', containmentGetter, []),
+                childrenMeta = {},
+                i;
+
+            for (i = 0; i < ruleHolders.length; i += 1) {
+                extendUniqueKeyedObject(childrenMeta, innerCore.getChildrenMeta(ruleHolders[i]));
+            }
+
+            if (Object.keys(childrenMeta).length === 0) {
+                return null;
+            }
+
+            return childrenMeta;
+        };
+
+        this.getPointerMeta = function (node, name) {
+            var ruleHolders = getAllMatchingRuleHolders(node, name, allValidRelationsNameGetter, []),
+                i,
+                pointerMeta = {};
+
+            if (ruleHolders.length === 0) {
+                return undefined;
+            }
+
+            for (i = 0; i < ruleHolders.length; i += 1) {
+                extendUniqueKeyedObject(pointerMeta, innerCore.getPointerMeta(ruleHolders[i], name));
+            }
+            return pointerMeta;
+        };
+
+        this.getValidTargetPaths = function (node, name) {
+            var getTargetPaths = function (getNode) {
```

Comment:
it is embedded because the function has to use the name input parameter of the outer scoped function, but cannot receive it as a parameter.
I can create a helper function that would return this function (but that would be a function definition within another one as well).
I am open for suggestions (the function is synchronous, so we could set a global 'name', but that would not be too elegant).